tasks/main.yml

```diff
@@ -17,26 +12,237 @@
+    - name: Check if the set edition matches the existing edition
+      shell: |
+        errorlog_edition="$(grep -oi '{{ mssql_edition }} edition' \
+        {{ __mssql_errorlog.stdout }})"
+        if [ -z "${errorlog_edition}" ]; then
+          edition_matches=false
+        else
+          edition_matches=true
+        fi
+        echo "${edition_matches}"
+      register: __mssql_edition_matches
+      ignore_errors: true
+      changed_when: false
+
+    - name: Ensure that the mssql-server service is stopped
+      service:
+        name: mssql-server
+        state: stopped
+      notify: Restart the mssql-server service
+      when: not __mssql_edition_matches.stdout | bool
+
+    - name: Change the edition of MSSQL
+      command: /opt/mssql/bin/mssql-conf set-edition
+      environment:
+        MSSQL_PID: "{{ mssql_edition }}"
+      register: __mssql_conf_set_edition
+      changed_when: '"The new edition is" in __mssql_conf_set_edition.stdout'
+      when: not __mssql_edition_matches.stdout | bool
+      notify: Restart the mssql-server service
+
+- name: Set the IP address setting for MSSQL
+  when:
+    - __mssql_conf_setup is skipped
+    - mssql_ip_address is defined
+  block:
+    - name: Check if the IP address setting is defined
+      command: /opt/mssql/bin/mssql-conf get network ipaddress
+      ignore_errors: true
+      changed_when: false
+      register: __mssql_conf_get_ip_address
+
+    - name: Change the IP address setting
+      command: >
+        /opt/mssql/bin/mssql-conf set network.ipaddress {{ mssql_ip_address }}
+      when: >
+        ("No setting for the given" in __mssql_conf_get_ip_address.stdout) or
+        (__mssql_conf_get_ip_address.stdout | regex_search('[^: ]*$')
+        not in mssql_ip_address)
+      notify: Restart the mssql-server service
+
+- name: Set the new TCP port for MSSQL
+  command: "/opt/mssql/bin/mssql-conf set network.tcpport {{ mssql_tcp_port }}"
+  register: __mssql_conf_set_tcp_port
+  ignore_errors: true
+  changed_when: >
+    "SQL Server needs to be restarted in order to apply this setting."
+    in __mssql_conf_set_tcp_port.stdout
+  notify: Restart the mssql-server service
+  when:
+    - __mssql_conf_setup is skipped
+    - mssql_tcp_port is defined
+
+- name: Input the {{ mssql_input_sql_file }} file to MSSQL
+  when: mssql_input_sql_file is defined
+  block:
+    - name: Create a tempfile on the host
+      tempfile:
+        state: file
+      register: __mssql_sql_tempfile
+
+    - name: Copy the {{ mssql_input_sql_file }} file to the host
+      copy:
+        src: "{{ mssql_input_sql_file }}"
+        dest: "{{ __mssql_sql_tempfile.path }}"
+        mode: preserve
+
+    - name: Ensure that the mssql-server service is started
+      service:
+        name: mssql-server
+        state: started
+
+    - name: Input the {{ mssql_input_sql_file }} file with the sqlcmd command
+      command: |
+        /opt/mssql-tools/bin/sqlcmd -U sa -P "{{ mssql_password }}"
```

Comment:
with `command` use `-P {{ mssql_password | quote }}` - I think that will allow you to accept passwords with `"` , `'`, `;`, and other shell metacharacters

Reply:
I just figured out that this log in query won't work because the role needs to check IP address and TCP port again, I will figure out how to do this next week.
As for your suggestion, `sqlcmd` does not parse passwords containing signle and double quotes and spaces. I see this as a limitation on the `sqlcmd` side, the best that the role can do about it is to tell users to not use those symbols in passwords, and use `| quote` as you suggested.
`mssql-conf` accepts passwords with those symbols, but `sqlcmd` cannot log in with such passwords.
For other prohibited symbols e.g. backslash `mssql-conf` tells that this symbol cannot be used with passwords.